app/models/developers/external_profiles/linkedin.rb

```diff
@@ -0,0 +1,54 @@
+module Developers::ExternalProfiles
+  class Linkedin
+    def initialize
+      @api_key = api_key
+      @endpoint = endpoint
+    end
+
+    def get_profile(url)
+      params = {
+        "url" => url,
+        "fallback_to_cache" => "on-error",
+        "use_cache" => "if-present"
+      }
+
+      conn = Faraday.new(@endpoint) do |faraday|
+        faraday.headers["Authorization"] = "Bearer #{@api_key}"
+        faraday.adapter Faraday.default_adapter
+        faraday.response :json
+      end
+
+      begin
+        response = conn.get("", params)
+        if response.status.to_i == 200
+          parse_json_response(response.body)
+        else
+          {error: "API Error: #{response.status} - #{response.body}"}
+        end
+      rescue Faraday::Error => e
+        {error: "Exception occurred: #{e.message}"}
+      end
+    end
+
+    private
+
+    def parse_json_response(response_body)
+      begin
+        # Attempt to access the first company name in the experiences array
+        {data: response_body["experiences"][0]}
+      rescue NoMethodError => e
+        {error: "JSON Parsing Error: #{e.message}"}
+      end
+    rescue JSON::ParserError => e
+      {error: "JSON Parsing Error: #{e.message}"}
+    end
```

Comment:
I think this has the same result, right?

```suggestion
    def parse_json_response(response_body)
      # Attempt to access the first company name in the experiences array
      {data: response_body["experiences"][0]}
    rescue JSON::ParserError, NoMethodError => e
      {error: "JSON Parsing Error: #{e.message}"}
    end
```